sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -0,0 +1,761 @@
+            .getInstance(TimeZone.getTimeZone(Constants.TIMEZONE_UTC.getValue()));
+
+
+    public static DefaultResultParameterProcessor getInstance() {
+        if (instance == null) {
+            synchronized (lock) {
+                if (instance == null) {
+                    instance = new DefaultResultParameterProcessor();
+                }
+            }
+        }
+        return instance;
+    }
+
+    protected BArray createAndPopulateBBRefValueArray(Object firstNonNullElement, Object[] dataArray,
+                                                      Type type) throws ApplicationError {
+        BArray refValueArray = null;
+        int length = dataArray.length;
+        if (firstNonNullElement instanceof String) {
+            refValueArray = createEmptyBBRefValueArray(PredefinedTypes.TYPE_STRING);
+            for (int i = 0; i < length; i++) {
+                refValueArray.add(i, dataArray[i]);
+            }
+        } else if (firstNonNullElement instanceof Boolean) {
+            refValueArray = createEmptyBBRefValueArray(PredefinedTypes.TYPE_BOOLEAN);
+            for (int i = 0; i < length; i++) {
+                refValueArray.add(i, dataArray[i]);
+            }
+        } else if (firstNonNullElement instanceof Integer) {
+            refValueArray = createEmptyBBRefValueArray(PredefinedTypes.TYPE_INT);
+            for (int i = 0; i < length; i++) {
+                refValueArray.add(i, dataArray[i]);
+            }
+        } else if (firstNonNullElement instanceof Long) {
+            refValueArray = createEmptyBBRefValueArray(PredefinedTypes.TYPE_INT);
+            for (int i = 0; i < length; i++) {
+                refValueArray.add(i, dataArray[i]);
+            }
+        } else if (firstNonNullElement instanceof Float) {
+            refValueArray = createEmptyBBRefValueArray(PredefinedTypes.TYPE_FLOAT);
+            for (int i = 0; i < length; i++) {
+                refValueArray.add(i, dataArray[i]);
+            }
+        } else if (firstNonNullElement instanceof Double) {
+            refValueArray = createEmptyBBRefValueArray(PredefinedTypes.TYPE_FLOAT);
+            for (int i = 0; i < length; i++) {
+                refValueArray.add(i, dataArray[i]);
+            }
+        } else if (firstNonNullElement instanceof BigDecimal) {
+            refValueArray = createEmptyBBRefValueArray(PredefinedTypes.TYPE_DECIMAL);
+            for (int i = 0; i < length; i++) {
+                refValueArray.add(i,
+                        dataArray[i] != null ? ValueCreator.createDecimalValue((BigDecimal) dataArray[i]) : null);
+            }
+        } else if (firstNonNullElement == null) {
+            refValueArray = createEmptyBBRefValueArray(type);
+            for (int i = 0; i < length; i++) {
+                refValueArray.add(i, firstNonNullElement);
+            }
+        } else {
+            createAndPopulateCustomBBRefValueArray(firstNonNullElement, dataArray, type);
```

Comment:
Shouldn't this be assigned to the `refValueArray` variable?
```
refValueArray = createAndPopulateCustomBBRefValueArray(firstNonNullElement, dataArray, type);
```

Reply:
Yes that should be changed.
We will update it

Reply:
Resolved with 0bd8dba16b48436cd248d7e02ae4cc046b33a2ff